src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2141,10 +2141,16 @@ private unsafe ParsingError PrivateParseMinimal()
                         _flags |= Flags.AuthorityFound;
                         idx += 2;
                     }
+                    // There is no Authority component, save the Path index
+                    // Note: mailto is the only guy who is treated specially, should be not.
                     else if (_syntax.NotAny(UriSyntaxFlags.MailToLikeUri))
                     {
-                        // There is no Authority component, save the Path index
-                        // Note: mailto is the only guy who is treated specially, should be not.
+                        // By now we know the URI has no Authority, so if the URI must be normalized, initialize it without one.
+                        if (_iriParsing && (_flags & Flags.HasUnicode) != 0 && (_flags & Flags.HostUnicodeNormalized) == 0)
+                        {
+                            _string = _string.Substring(0, idx);
```

Comment:
In what situations do we now incur this extra string allocation?

Reply:
We'll hit this code path only when we have a URI that contains Unicode, has a scheme that is not followed by "//" or "///",  and does not have special host parsing (file paths, mailto). This is the same operation as is performed for all other URIs with Unicode. It's just that we missed cutting out everything but the scheme just in this specific case.

If you have any thoughts on how to make the operation more efficient, I'd definitely be open to that though!

Reply:
@stephentoub are you satisfied with the expected performance impact here? If so I'm going to go ahead and merge this.

Reply:
Ok, thanks.